pkg/mimirpb/timeseries.go

```diff
@@ -447,6 +448,16 @@ func ReuseTimeseries(ts *TimeSeries) {
 
 // ClearExemplars safely removes exemplars from TimeSeries.
 func ClearExemplars(ts *TimeSeries) {
+	// When cleaning exemplars, retain the slice only if its capacity is not bigger than
+	// the desired max preallocated size. This allow us to ensure we don't put very large
+	// slices back to the pool (e.g. a few requests with an huge number of exemplars may cause
+	// in-use heap memory to significantly increase, because the slices allocated by such poison
+	// requests would be reused by other requests with a normal number of exemplars).
+	if cap(ts.Exemplars) > maxPreallocatedExemplarsPerSeries {
+		ts.Exemplars = nil
```

Comment:
I'm not an expert on GC. I'm wondering if this will for sure let the GC now it can garbage collect the former big slice.

Because if the GC won't garbage collect it right away then we might want to set this if below the next for that makes sure name and value of exemplars don't point to any large grpc buffers.

Reply:
The slice will be GCed at the next GC run.

Reply:
Yep just that its enough, I guess the slice is not referenced anymore anywhere so next run gcs it 👍  Thanks just wanted to make sure of it.